finagle-thrift/src/main/scala/com/twitter/finagle/thrift/ThriftClientBufferedCodec.scala

```diff
@@ -40,13 +40,15 @@ object ThriftClientBufferedCodec {
     apply()
   }
 
-  def apply(protocolFactory: TProtocolFactory) =
-    new ThriftClientBufferedCodecFactory(protocolFactory)
+  def apply(protocolFactory: TProtocolFactory, _attemptProtocolUpgrade: Boolean = true) =
```

Comment:
for backwards compatibility with java users (who don't get the benefits of default arguments) can you add it as forwarding method instead:

```
def apply(protocolFactory: TProtocolFactory) =
  apply(protocolFactory, true)

def apply(protocolFactory: TProtocolFactory, _attemptProtocolUpgrade: Boolean) =
  ...
```

also, since you are editing, can you add return types?